rixs/process2d.py

```diff
@@ -0,0 +1,550 @@
+    y_centers = (y_edges[:-1] + y_edges[1:])/2
+    x_centers = (x_edges[:-1] + x_edges[1:])/2
+
+    ref_column = image[:, image.shape[1]//2]
+
+    offsets = np.array([])
+    for column in image.T:
+        cross_correlation = np.correlate(column, ref_column, mode='same')
+        offsets = np.append(offsets, y_centers[np.argmax(cross_correlation)])
+    return x_centers, offsets - offsets[offsets.shape[0]//2]
+
+
+def fit_curvature(photon_events, bins=None,
+                  guess_curvature={'x2': 0., 'x1': 0., 'x0': 500.},
+                  vary_params={'x2': True, 'x1': True, 'x0': True}):
+    """Get offsets, fit them and return polynomial that defines the curvature
+
+    Parameters
+    -------------
+    photon_events : array
+        two column x, y photon location coordinates
+    bins : int or array_like or [int, int] or [array, array]
+        The bin specification in y then x order:
+            * If int, the number of bins for the two dimensions (nx=ny=bins).
+            * If array_like, the bin edges for the two dimensions
+              (y_edges=x_edges=bins).
+            * If [int, int], the number of bins in each dimension
+              (ny, nx = bins).
+            * If [array, array], the bin edges in each dimension
+              (y_edges, x_edges = bins).
+            * A combination [int, array] or [array, int], where int
+              is the number of bins and array is the bin edges.
+    curvature : dictionary
+    n2d order polynominal coeficients defining
+    curvature
+    {'x^2': coef, 'x1': coef, 'x0': coef}
+
+    Returns
+    -----------
+
+    """
+    x_centers, offsets = get_curvature_offsets(photon_events, bins)
+    poly_model = lmfit.Model(poly, independent_vars=['x'])
+    params = poly_model.make_params()
+    for name, c in guess_curvature.items():
+        params[name].set(value=c)
+    for name, TF in vary_params.items():
+        params[name].set(value=TF)
+    result = poly_model.fit(offsets, x=x_centers, params=params)
+
+    if result.success is not True:
+        print(result.message)
+
+    return result.best_values
+
+
+def plot_curvature(ax1, curvature, photon_events, plot_offset=0.):
+    """ Plot a red line defining curvature on ax1
+
+    Parameters
+    ----------
+    ax1 : matplotlib axes object
+        axes for plotting on
+    curvature : dictionary
+        n2d order polynominal coeficients defining image curvature
+        {'x^2': coef, 'x1': coef, 'x0': coef}
+    photon_events : array
+        two column x, y photon location coordinates
+
+    Returns
+    ---------
+    curvature_artist : matplotlib artist object
+        artist from image scatter plot
+    """
+    x = np.arange(np.nanmax(photon_events[:, 0]))
+    y = poly(x, **curvature)
+    return plt.plot(x, y + plot_offset, 'r-')
+
+
+def estimate_elastic_pos(photon_events, x_range=(0, 20), bins=None):
+    """
+    Estimate where the elastic line.
+
```

Comment:
Can we expand the description here to include the fact that you are assuming a 'peak' in x_range and centering based on the maximum intensity?

It also requires a description of the returns (which is missing)